### PR TITLE
Interview assignment

### PR DESCRIPTION
**Pull request by shail3sh**
Hi Grady
Please have a look at the stuff I could complete. I spent roughly 8 hours on this...it took longer because I haven't coded in rails for last couple of years. But I was still able to finish a lot of functionalities. Could not get to writing tests. So please focus on the functionalities only. If you need me to focus on tests, I can work on it and send another pull request.

### CODE REVIEW

**Comment by glaksmono** [comment]
Here are some of my general comments:
1. Doesn't seem like you validates the Email address in the correct format?
2. It allows the password to be less than 6 characters (not in the requirement, but doesn't seem to be a good practice)
3. Need tests [MOKA-003]
4. It'd be great if you can do extra credit [MOKA-004]

File: app/models/user.rb [hunk 1]
```diff
@@ -1,10 +1,25 @@
 class User < ActiveRecord::Base
-
   has_secure_password
 
+  attr_accessor :current_password
```
Comment (glaksmono) [inline app/models/user.rb]: Why do we need this? Doesn't seem like a good practice?

File: app/models/user.rb [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
 
   validates :first_name,  presence: true
   validates :last_name,  presence: true
   validates :email, presence: true
+
+  def full_name
+    (first_name || "") + " " + (last_name || "")
+  end
+
+  validate :validates_current_password
```
Comment (glaksmono) [inline app/models/user.rb]: Style - Should we move all of the `one-line functions` at the top of the class?

File: config/database.yml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 development: &local
   adapter: postgresql
-  database: mokatest
+  database: mokadev
   host: localhost
-  username: postgres
-  password: 12345678
+  username: shail3sh
```
Comment (glaksmono) [inline config/database.yml:5]: Style - should this be part of `.gitignore`?

File: config/environments/test.rb [hunk 4]
```diff
@@ -39,4 +39,10 @@
 
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
+
+  # Enable stdout logger
+  config.logger = Logger.new(STDOUT)
```
Comment (glaksmono) [inline config/environments/test.rb]: What is this for? Is this `diff` necessary?

**Comment by glaksmono** [comment]
@shail3sh thoughts about this?

```
➜  moka-rails-interview git:(master) ✗ rspec test/models/
No examples found.


Finished in 0.00054 seconds (files took 0.0803 seconds to load)
0 examples, 0 failures

➜  moka-rails-interview git:(master) ✗ rspec test/controllers
No examples found.


Finished in 0.00029 seconds (files took 0.05876 seconds to load)
0 examples, 0 failures

➜  moka-rails-interview git:(master) ✗
```

File: app/models/user.rb (hunk 1 above)
Comment (shail3sh) [reply app/models/user.rb]: It was required when the profile edit update action is triggered and led to model update.

File: app/models/user.rb (hunk 2 above)
Comment (shail3sh) [reply app/models/user.rb]: done

File: config/database.yml (hunk 3 above)
Comment (shail3sh) [reply config/database.yml:5]: Yes, it should be part of .gitignore, done.

File: config/environments/test.rb (hunk 4 above)
Comment (shail3sh) [reply config/environments/test.rb]: This I forgot to remove. Done.

**Pull request closed without merging**